src/middleware/router.coffee

```diff
@@ -151,6 +161,11 @@ sendRequestToRoutes = (ctx, routes, next) ->
               # handle mediator reponse
               responseObj = JSON.parse response.body
               ctx.mediatorResponse = responseObj
+
+              if responseObj.error?
+                ctx.autoRetry = true
```

Comment:
It gets picked up here to set flag the transaction: https://github.com/jembi/openhim-core-js/blob/8f892848ef2ed9eb4b851db73c2340b450a6f008/src/middleware/messageStore.coffee#L168